src/Roslyn.Diagnostics.Analyzers/Core/ExportedPartsShouldHaveImportingConstructorCodeFixProvider.cs

```diff
@@ -0,0 +1,189 @@
+            {
+                declaration = generator.GetDeclaration(declaration.Parent);
+                if (declaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(declaration);
+            }
+
+            var newDeclaration = generator.WithAccessibility(declaration, Accessibility.Public);
+            return document.WithSyntaxRoot(root.ReplaceNode(declaration, newDeclaration));
+        }
+
+        private async Task<Document> AddImportingConstructorAttributeAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var constructor = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var declaration = constructor;
+            var declarationKind = generator.GetDeclarationKind(declaration);
+            while (declarationKind != DeclarationKind.Constructor)
+            {
+                declaration = generator.GetDeclaration(declaration.Parent);
+                if (declaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(declaration);
+            }
+
+            var exportedType = semanticModel.GetDeclaredSymbol(declaration, cancellationToken)?.ContainingType;
+            if (exportedType is null)
+            {
+                return document;
+            }
+
+            INamedTypeSymbol importingConstructorAttributeSymbol = null;
+            foreach (var attributeData in exportedType.GetAttributes())
+            {
+                INamedTypeSymbol exportAttributeSymbol = null;
+                foreach (var attributeClass in attributeData.AttributeClass.GetBaseTypesAndThis())
+                {
+                    if (attributeClass.Name == nameof(ExportAttribute))
+                    {
+                        exportAttributeSymbol = attributeClass;
+                        break;
+                    }
+                }
+
+                if (exportAttributeSymbol is null)
+                {
+                    continue;
+                }
+
+                importingConstructorAttributeSymbol = exportAttributeSymbol.ContainingNamespace.GetTypeMembers(nameof(ImportingConstructorAttribute)).FirstOrDefault();
+                if (importingConstructorAttributeSymbol is object)
+                {
+                    break;
+                }
+            }
+
+            var newDeclaration = generator.AddAttributes(declaration, generator.Attribute(generator.TypeExpression(importingConstructorAttributeSymbol)));
```

Comment:
Bail out on null `importingConstructorAttributeSymbol`?

Reply:
Will add handling for this in the analyzer instead